internal/telemetry/client_test.go

```diff
@@ -282,96 +276,40 @@ func TestAgentlessRetry(t *testing.T) {
 	}))
 	brokenServer.Close()
 
-	client := &telemetry.Client{
+	client := &Client{
 		URL: brokenServer.URL,
 	}
-	client.Start([]telemetry.Configuration{})
+	client.Start(nil)
 	waitAgentlessEndpoint()
 }
 
 func TestCollectDependencies(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
-	received := make(chan *telemetry.Dependencies)
+	received := make(chan *Dependencies)
 
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Header.Get("DD-Telemetry-Request-Type") == string(telemetry.RequestTypeDependenciesLoaded) {
-			var body telemetry.Body
-			body.Payload = new(telemetry.Dependencies)
+		if r.Header.Get("DD-Telemetry-Request-Type") == string(RequestTypeDependenciesLoaded) {
+			var body Body
+			body.Payload = new(Dependencies)
 			err := json.NewDecoder(r.Body).Decode(&body)
 			if err != nil {
 				t.Errorf("bad body: %s", err)
 			}
 			select {
-			case received <- body.Payload.(*telemetry.Dependencies):
+			case received <- body.Payload.(*Dependencies):
 			default:
 			}
 		}
 	}))
 	defer server.Close()
-	client := &telemetry.Client{
+	client := &Client{
 		URL: server.URL,
 	}
-	client.Start([]telemetry.Configuration{})
+	client.Start(nil)
 	select {
 	case <-received:
 	case <-ctx.Done():
 		t.Fatalf("Timed out waiting for dependency payload")
 	}
 }
-
-func TestProductChange(t *testing.T) {
```

Comment:
moving this to the `telemetry_test.go` file, and changing the test to directly assert on the Client's `requests` slice so we don't have to rely on heartbeat and flushing